### ISSUE TITLE
Common Quarkus project marked as duplicate classes in tests (split package issue)

### ISSUE THREAD

**Issue by brunohpg** [quarkusio/quarkus#21953]
### Describe the bug

Hi,

In my ecosystem, I have several "main projects" and one "common project".

### Configuration

All projects, including the common one, are Quarkus projects.
The common project is depdency of the main projects.
I use Gradle, so in the main projects I have it in build.gradle:

`implementation project(':flexi-common')`

And settings.gradle:

```
include ':flexi-common'
project(':flexi-common').projectDir = new File(settingsDir, '../flexi-common')
```

For correct indexing, all main projects have in "applications.properties":

```
quarkus.index-dependency.flexi-common.group-id=br.com.grupomasut.flexi
quarkus.index-dependency.flexi-common.artifact-id=flexi-common
```

The common project cannot add new properties using "applications.properties", so it is blank.

### Execution

Projects work correctly with dependency injection when they are put into production. 
However, they have problems when the tests are run (`quarkusTest`).

I created a test in the common project and I got the errors:

```
2021-12-06 09:48:46,018 WARN  [io.qua.arc.dep.SplitPackageProcessor] (build-32) Detected a split package usage which is considered a bad practice and should be avoided. Following packages were detected in multiple archives: 
- "br.com.grupomasut.flexi.common.rest" found in [application classes, br.com.grupomasut.flexi:flexi-common::jar]
- "br.com.grupomasut.flexi.common.rastreamento.vcar.service" found in [application classes, br.com.grupomasut.flexi:flexi-common::jar]
...
```

```
2021-12-06 09:48:46,396 ERROR [io.qua.test] (Test runner thread) ==================== TEST REPORT #1 ====================
2021-12-06 09:48:46,396 ERROR [io.qua.test] (Test runner thread) Test VeiculosRemoteServiceTest#converterPlaca() failed 
: java.lang.RuntimeException: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.SyntheticBeansProcessor#initStatic threw an exception: java.lang.IllegalStateException: A synthetic bean with identifier 1b62793a0fe03e7a4b59c9c159a923b0b6453b94 is already registered: SYNTHETIC bean [types=[br.com.grupomasut.flexi.common.rastreamento.vcar.graphql.VcarGraphqlClient], qualifiers=[@Default, @Any], target=n/a]
	at io.quarkus.arc.processor.BeanDeployment.addSyntheticBean(BeanDeployment.java:1165)
...
```

### Chaning Configurations

I don't understand the reason for the "duplication" of packages. At least that's what it looks like, the common project is entered twice.
But I suppose the gradle + applications.properties are including the common project twice.

So I changed Gradle's settings to "compileOnly", "compile", "runtime", but don't work.
In some cases the error persists in others it is not possible to resolve the dependency injections in the code.

I commented on indexing in "applications.properties", also without success.

It is also possible to bypass the "split package" problem by adding in "applications.properties" of the common project:

`quarkus.arc.ignored-split-packages=br.com.grupomasut.flexi.*`

But apparently it only masks the problem.

### Expected behavior

I expected the tests to run successfully on both the common project and the main project.

### Actual behavior

Execution proceeds as expected. However, the tests general the cited errors.

### How to Reproduce?

Create two Quarkus projects:

- Project 1 (lib)
 - Write a test
- Project 2 (main)
 - Include Project 1 as dependency

Try quarkusTest in Project 2

### Output of `uname -a` or `ver`

Microsoft Windows [versão 10.0.19043.1348]

### Output of `java -version`

azul-15.0.4

### Quarkus version or git rev

2.5.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 6.9

**Comment by glefloch** [comment]
thanks for reporting this @brunohpg. We have several sample projects used in our test suite that leverage multi-module project such as : https://github.com/quarkusio/quarkus/tree/main/integration-tests/gradle/src/main/resources/multi-module-named-injection
does the `quarkusDev` task works? could you create a simple reproducer?

**Comment by brunohpg** [comment]
@glefloch 

Yes, the `quarkusDev` works properly.

I updated all projects to version 5.6.1. I had to make some additional configurations in the common project, as before there was no database configured and now it is mandatory, or the test will try to start a docker container.

Looks like everything worked fine after that. I don't know if I did something wrong before.

However, there is a difference from the example project you sent: my test was in the common project, not the main project.

Anyway, thanks. In the last version everything worked. I will close the call.
